### ISSUE TITLE
Accuracy remain the same, just like random guess

### ISSUE THREAD

**Issue by yzloveyz** [BVLC/caffe#2430]
Hi all,  

I was running Caffe on my own dataset, which is a 8-class classification problem. My configuration is like the 8-layer model from "bvlc_reference_caffenet", only changing the training batch size to 128 and validation batch size to 50 because of power issue. 

**Now my problem is the accuracy of the network remains the same (12.5%) since the beginning, which is random guessing for a 8-class problem. I think the network is not learning anything.** 

I0507 19:42:05.711138 11638 solver.cpp:189] Iteration 7980, loss = 2.07975
I0507 19:42:05.711185 11638 solver.cpp:204]     Train net output #0: loss = 2.07975 (* 1 = 2.07975 loss)
I0507 19:42:05.711200 11638 solver.cpp:464] Iteration 7980, lr = 0.01
I0507 19:42:10.959509 11638 solver.cpp:266] Iteration 8000, Testing net (#0)
I0507 19:43:35.754755 11638 solver.cpp:315]     **Test net output #0: accuracy = 0.12484**
I0507 19:43:35.754869 11638 solver.cpp:315]     Test net output #1: loss = 2.35999 (* 1 = 2.35999 loss)
I0507 19:43:35.850280 11638 solver.cpp:189] Iteration 8000, loss = 2.07875
I0507 19:43:35.850329 11638 solver.cpp:204]     Train net output #0: loss = 2.07875 (* 1 = 2.07875 loss)
I0507 19:43:35.850345 11638 solver.cpp:464] Iteration 8000, lr = 0.01

However, when I use t-SNE to visualize my dataset as below, it seems like it should be separable. 
![caffeissue](https://cloud.githubusercontent.com/assets/11484913/7529612/28c036ec-f4f0-11e4-8929-ecf942f2292f.png)

But right now, the Caffe just doesn't work. So if any of you came across this kind of problem before, please share some ideas. Thank you so much. 

Yi

**Comment by longjon** [comment]
Please ask usage and modeling question on the [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) list. Thanks!

**Comment by sayadyaghoobi** [comment]
i faced the same problem, if you solved that, please share your ways. many thanks

**Comment by PENGUINLIONG** [comment]
Some possible solutions:

1. Try explicitly set your loss layer's weight with `loss_weight: 1`. So the layer is forced to back-propagate.
```
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "fc"
  bottom: "label"
  top: "loss"
  loss_weight: 1
}
```
2. Recompile your Caffe without cuDNN. It's heard of that cuDNN sometimes causes training problems.

**Comment by sayadyaghoobi** [comment]
@PENGUINLIONG  many thanks, i test this, i'll show my results
